api/controllers/UserController.js

```diff
@@ -29,11 +29,30 @@
 
 module.exports = {
 
+  update: function(req, res) {
+    let userData = JsonApiService.deserialize(req.body.data);
+    if (req.user.isAdmin || req.user.id === req.allParams().id) {
+      return User.update({
```

Comment:
You can probably replace this with `return JsonApiService.updateOneRecord(req, res);`
Once authentication is verified, it's ok to juste pass date to the blueprint.